javaslang/generator/Generator.scala

```diff
@@ -1265,46 +1265,40 @@ def generateMainClasses(): Unit = {
 
             ${(i == 0).gen(xs"""
               @Override
-              public <T> Tuple1<T> prepend(T value) {
+              public <T> Tuple1<T> append(T value) {
                  return new Tuple1<>(value);
               }
 
               @Override
-              public <T> Tuple1<T> append(T value) {
+              public <T> Tuple1<T> prepend(T value) {
                  return new Tuple1<>(value);
               }
             """)}
 
             ${(i > 0 && i < N).gen(xs"""
               @Override
-              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
-                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
-              }
-
-              @Override
               public <T> Tuple${i + 1}<${(1 to i).gen(j => s"T$j")(", ")}, T> append(T value) {
                   return Tuple.of(${(1 to i).gen(j => s"_$j")(", ")}, value);
               }
-            """)}
 
-            ${(i == N).gen(xs"""
               @Override
-              public <T> Tuple prepend(T value) {
-                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
+              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
+                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
               }
+            """)}
 
+            ${(i == N).gen(xs"""
               @Override
               public <T> Tuple append(T value) {
                   throw new UnsupportedOperationException("Append to Tuple${N}");
               }
-            """)}
 
-            ${(i > 0).gen(xs"""
-              public static $generics $className<${(1 to i).gen(j => s"Seq<? extends T$j>")(", ")}> sequence(Iterable<$className<${(1 to i).gen(j => s"? extends T$j")(", ")}>> tuples) {
-                Objects.requireNonNull(tuples, "tuples is null");
-                return new $className<>(${(1 to i).gen(j => s"${im.getType("javaslang.collection.Iterator")}.ofAll(tuples).map($className::_$j).toList()")(", ")});
+              @Override
+              public <T> Tuple prepend(T value) {
+                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
```

Comment:
A `list.toTuple()` is not possible because the number of elements may exceed the max Tuple arity.

`tuple.toSeq()` exists already.

What we really need is a heterogenous list (`HList`). We have already an issue for that but the generics are not human-readable any more when the list grows. For now I would not implement an HList. It would start to make sense when we have local variable type inference in Java.

@ashrko619 Sorry for being torn back and forth. I think we need a more in-depth analysis for future features, including code examples, pros, cons etc.

Please give me some time to think about this feature...